pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -1847,4 +1867,88 @@ mod tests {
             }
         );
     }
+
+    async fn inject_index_part(test_state: &TestSetup, generation: Generation) -> IndexPart {
+        // An empty IndexPart, just sufficient to ensure deserialization will succeed
+        let example_index_part = IndexPart::new(
+            HashMap::new(),
+            "0/16960E8".parse::<Lsn>().unwrap(),
+            TimelineMetadata::example(),
```

Comment:
Please don't repeat the `example()`s LSN here. 
Either use my suggestion or add an `IndexPart::example()` method?

```suggestion
        let example_metadata = TimelineMetadata::example();
        let example_index_part = IndexPart::new(
            HashMap::new(),
            example_metadata.disk_consistent_lsn(),
            example_metadata,
            
```